File: frontend/src/lib/dtos/cadaster/cadaster.dto.ts
export type ForestStandListDto = {
	id: string;
	number: number;
	area: number;
	totalVolume: number;
	isActive: boolean;
};

export type ActivityListDto = {
	id: string;
	description: string;
	quantity: number;
	unit: string | null;
	date: string;
	activityTypeName: string;
	userName: string;
	userFirstName: string | null;
	userLastName: string | null;
	cadasterCadastralNumber: string | null;
	forestStandNumber: number;
	locationDescription: string | null;
	applicationStatus: string | null;
};

export type RecentActivityDto = {
	id: string;
	description: string;
	quantity: number;
	unit: string | null;
	date: string;
	activityTypeName: string;
	userName: string;
	userFirstName: string | null;
	userLastName: string | null;
	cadasterCadastralNumber: string | null;
	forestStandNumber: number;
};

export type CadasterDto = {
	id: string;
	cadastralNumber: string;
	forestArea: number;
	arableArea: number;
	grasslandArea: number;
	yardArea: number;
	buildingFootprintArea: number;
	underwaterArea: number;
	otherArea: number;
	soilQualityIndex: number;
	calculatedVolume: number;
	volumeGrowth: number;
	landPropertyId: string;
	landPropertyName: string;
	landPropertyIsFsc: boolean;
	forestStands: ForestStandListDto[];
	recentActivities: RecentActivityDto[];
};

export type CadasterUpdateDto = {
	id: string;
	cadastralNumber: string;
	forestArea: number;
	arableArea: number;
	grasslandArea: number;
	yardArea: number;
	buildingFootprintArea: number;
	underwaterArea: number;
	otherArea: number;
	soilQualityIndex: number;
	calculatedVolume: number;
	volumeGrowth: number;
	landPropertyId: string;
};

File: ForestManagement/App.DTO/Cadaster/CadasterDto.cs
using App.DTO.ForestStand;

namespace App.DTO.Cadaster;

/// <summary>
/// Full cadaster details response DTO with forest stands and recent activities
/// </summary>
public class CadasterDto
{
    public Guid Id { get; set; }
    public string CadastralNumber { get; set; } = string.Empty;
    public decimal? ForestArea { get; set; }
    public decimal? ArableArea { get; set; }
    public decimal? GrasslandArea { get; set; }
    public decimal? YardArea { get; set; }
    public decimal? BuildingFootprintArea { get; set; }
    public decimal? UnderwaterArea { get; set; }
    public decimal? OtherArea { get; set; }
    public int? SoilQualityIndex { get; set; }
    public int? CalculatedVolume { get; set; }
    public decimal? VolumeGrowth { get; set; }
    public Guid LandPropertyId { get; set; }
    public string LandPropertyName { get; set; } = string.Empty;
    public bool LandPropertyIsFsc { get; set; }
    
    // Related forest stands
    public ICollection<ForestStandListDto> ForestStands { get; set; } = new List<ForestStandListDto>();
    
    // Recent activities (5 most recent)
    public ICollection<Activity.RecentActivityDto> RecentActivities { get; set; } = new List<Activity.RecentActivityDto>();
}
